src/main/java/digital/slovensko/autogram/core/SigningJob.java

```diff
@@ -180,47 +181,50 @@ private static SigningJob build(DSSDocument document, SigningParameters params,
         var level = SignatureValidator.getSignedDocumentSignatureLevel(SignatureValidator.getSignedDocumentSimpleReport(document));
         if (level != null) switch (level.getSignatureForm()) {
             case PAdES:
-                return SigningParameters.buildForPDF(document, checkPDFACompliance, isEn319132, tspSource);
+                return SigningParameters.buildForPDF(document.getDocument(), checkPDFACompliance, isEn319132, tspSource);
             case XAdES:
-                return SigningParameters.buildForASiCWithXAdES(document, checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
+                return SigningParameters.buildForASiCWithXAdES(document.getDocument(), checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
             case CAdES:
-                return SigningParameters.buildForASiCWithCAdES(document, checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
+                return SigningParameters.buildForASiCWithCAdES(document.getDocument(), checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
             default:
                 ;
         }
 
-        if (isPDF(document.getMimeType())) switch (signatureType) {
+        if (isPDF(document.getDocument().getMimeType())) switch (signatureType) {
             case PAdES_BASELINE_B:
-                return SigningParameters.buildForPDF(document, checkPDFACompliance, isEn319132, tspSource);
+                return SigningParameters.buildForPDF(document.getDocument(), checkPDFACompliance, isEn319132, tspSource);
             case XAdES_BASELINE_B:
-                return SigningParameters.buildForASiCWithXAdES(document, checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
+                return SigningParameters.buildForASiCWithXAdES(document.getDocument(), checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
             case CAdES_BASELINE_B:
-                return SigningParameters.buildForASiCWithCAdES(document, checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
+                return SigningParameters.buildForASiCWithCAdES(document.getDocument(), checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
             default:
                 ;
         }
 
-        return SigningParameters.buildForASiCWithXAdES(document, checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
+        return SigningParameters.buildForASiCWithXAdES(document.getDocument(), checkPDFACompliance, isEn319132, tspSource, plainXmlEnabled);
     }
 
     public boolean shouldCheckPDFCompliance() {
-        return parameters.getCheckPDFACompliance() && isPDF(document.getMimeType());
+        // PDF/A doesn't support encryption
+        return parameters.getCheckPDFACompliance() && isPDF(document.getDocument().getMimeType()) && document.getOpenDocumentPassword().length == 0;
```

Comment:
Rozumiem, prečo si to chcel, ale v skutočnosti chceme opak - ak má user zapnutú kontrolu PDF/A a príde mu dokuemntu, ktorý nie je PDF/A, chce vidieť warning. Tuto hovoríš, že pri encrypted dokumentoch to nikdy nie je PDF/A `a preto ani nezobrazíš warning`. Práveže vtedy by sa mal zobraziť tiež.

`Autogram.java:60` potom hodí exception z `new PDFAStructureValidator().validate`, takže pred to môžeš dať check, či je tam heslo a v tom prípade rovno vyvolať warning aj bez ďalšej PDF/A validácie. (ten check na existujúce heslo zabaľ do nejakej metódy priamo v AutogramDocument